plaso/parsers/cookie_plugins/ganalytics.py

```diff
@@ -410,29 +410,30 @@ def GetEntries(
       for variable in extra_variables:
         key, _, value = variable.partition('=')
 
-        # Cookies can have a variety of different encodings, usually ASCII or
-        # UTF-8, and values may additionally be URL encoded. urllib only
-        # correctly url-decodes ASCII strings, so we'll convert our string
-        # to ASCII first.
-        try:
-          ascii_value = value.encode('ascii')
-        except UnicodeEncodeError:
-          ascii_value = value.encode('ascii', errors='replace')
-          parser_mediator.ProduceExtractionError(
-              'Cookie contains non 7-bit ASCII characters, which have been '
-              'replaced with a "?".')
-
-        utf_stream = urlparse.unquote(ascii_value)
-
-        try:
-          value_line = utf_stream.decode('utf-8')
-        except UnicodeDecodeError:
-          value_line = utf_stream.decode('utf-8', errors='replace')
-          parser_mediator.ProduceExtractionError(
-              'Cookie value did not decode to Unicode string. Non UTF-8 '
-              'characters have been replaced.')
-
-        extra_attributes[key] = value_line
+        # Urllib2 in Python2 requires a str argument, not unicode. We thus need
+        # to convert the 'value' argument to str and back again, but only in
+        # Python 2.
+        if isinstance(value, py2to3.UNICODE_TYPE) and py2to3.PY_2:
+          try:
+            value = codecs.decode(value, 'ascii')
+          except UnicodeEncodeError:
+            value = codecs.decode(value, 'ascii', errors='replace')
+            parser_mediator.ProduceExtractionError(
+                'Cookie contains non 7-bit ASCII characters, which have been '
+                'replaced with a "?".')
+
+        value = urlparse.unquote(value)
+
+        if py2to3.PY_2:
+          try:
+            value = codecs.encode(value, 'utf-8')
+          except UnicodeDecodeError:
+            value = codecs.encode(value, 'utf-8', errors='replace')
+            parser_mediator.ProduceExtractionError(
+                'Cookie value did not decode to Unicode string. Non UTF-8 '
```

Comment:
decode => encode ?

Reply:
Changed this. decode/encode is a little confusing here.